### DISCUSSION TITLE
Why does PHP image scan rely on composer.lock instead of installed.json?

### DISCUSSION THREAD

**Discussion by namandf** [aquasecurity/trivy#5398] [Q&A]
### Question

It's not necessary that one would copy over the composer.lock file into the container. How would trivy detect the SBOM in such cases?

The composer typically creates a vendor folder with all installed dependencies  and also an installed.json file which provides insights into installed packages and metadata. Any specific reason for not using it?

### Target

Container Image

### Scanner

Vulnerability

### Output Format

Table

### Mode

Standalone

### Operating System

macOS

### Version

```bash
Version: 0.44.1
Vulnerability DB:
  Version: 2
  UpdatedAt: 2023-10-05 00:18:15.319890951 +0000 UTC
  NextUpdate: 2023-10-05 06:18:15.319890451 +0000 UTC
  DownloadedAt: 2023-10-05 05:46:56.958307 +0000 UTC
Java DB:
  Version: 1
  UpdatedAt: 2023-09-11 00:53:00.064262708 +0000 UTC
  NextUpdate: 2023-09-14 00:53:00.064262008 +0000 UTC
  DownloadedAt: 2023-09-11 19:22:04.868176 +0000 UTC
```

**Comment by knqyf263** [comment]
In progress
https://github.com/aquasecurity/trivy/pull/4865

**Reply by knqyf263** [reply]
Here is the issue.
https://github.com/aquasecurity/trivy/issues/5057
